### ISSUE TITLE
RedisSlidingWindowRateLimiter does not provide IdleDuration

### ISSUE THREAD

**Issue by hacst** [cristipufu/aspnetcore-redis-rate-limiting#61]
The current implementation always returns `TimeSpan.Zero` for [RateLimiter.IdleDuration](https://learn.microsoft.com/en-us/dotnet/api/system.threading.ratelimiting.ratelimiter.idleduration?view=aspnetcore-7.0#system-threading-ratelimiting-ratelimiter-idleduration). When using a partitioned rate limiter this means the rate limiters created for the partitions [never get cleaned up](https://github.com/dotnet/runtime/blob/c3859ad70b3d5d02b252d07c904160eada32d653/src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/DefaultPartitionedRateLimiter.cs#LL229C17-L229C17) and will accumulate.

I think this is true for all rate limiters in this repo however as it is the only one I am using right now, I only looked at the sliding window rate limiter in detail so far. For that the most reasonable approach to me seems to be to accept some imprecision and just track the last `expireat` value set in the `RedisSlidingWindowManager` and provide an estimated idle duration for that. This is under the assumption that if a manager ever deletes a partition that was not actually fully idle, it is really no problem to have it just be re-created by the factory. The actual limit is still in redis after all.

I could create a pull request for the redis sliding rate limiter if this approach is acceptable.

This would not fully match what the interface wants though. The documentation says:
> Specifies how long the [RateLimiter](https://learn.microsoft.com/en-us/dotnet/api/system.threading.ratelimiting.ratelimiter?view=aspnetcore-7.0) has had all permits available.

A better estimate or precise answer could of course be given by reaching out to redis, however to perform potentially blocking operations in there seems problematic to me.

**Comment by dlxeon** [comment]
I faced same memory issue with `RedisTokenBucketRateLimiter`. I use client IP address as partition key, with 100k unique IP addresses per day, that quickly grows not only in memory utilization, but also CPU utilization.
![image](https://github.com/cristipufu/aspnetcore-redis-rate-limiting/assets/14337206/f240704b-25dc-473e-9b21-50fd16842f3d)

As as solution I subclassed `RedisTokenBucketRateLimiter`, overriden AcquireAsyncCore and IdleDuration to return time passed since last usage of limiter.
